Use cleanWhiteSpace to delete multiple spaces between words

TongJ05 opened the pull request:
**Related Issue:**
closes #416 
**Description of Changes**
Create a separate method to clean spaces in the beginning, at the end of a word, or multiple spaces between words and use this method instead of trim() on inputs.
<!-- In 1-3 sentences, provide an overview of what changes were made and why. -->

TongJ05 committed:
Use cleanWhiteSpace to delete multiple spaces between words

vadekh reviewed (changes requested):
This will not work until the new `cleanWhitespace` function is imported into all of the other files that use it. It should be an easy change, most of the files already import other utilities so just copy that format.

dwhieb reviewed (changes requested):
The way `cleanWhiteSpace` is currently written, it needs to have the string passed to it, instead of being chained on the end of the string:

❌ `string.cleanWhiteSpace()`
✅ `cleanWhiteSpace(string)`

TongJ05 committed:
Merge branch 'main' into issue-416

TongJ05 committed:
Fix cleanWhitespace and add tests to test it

Use cleanWhitespace() in the correct way and make it to replace tabs with a single whitespace. Also add unit tests to test cleanWhitespace().

dwhieb committed:
clean whitespace on language name

dwhieb reviewed (dismissed):
Looks good! Made one minor correction (`cleanWhitespace` was being called on the wrong variable), and also added a spot I missed previously (the language name). I checked each of the fields in the Language Editor on my machine by typing in tabs or multiple spaces to see if they saved correctly, and they did.

dwhieb committed:
Update src/pages/Languages/LanguageEditor/LanguageEditor.js
